next-docs/pages/world/systems.mdx

```diff
@@ -1 +1,111 @@
+If it doesn't exist yet it is created, and the address that called `registerSystem` is assigned as the owner of the new namespace.
+
+In a `World` there can only be one `System` per contract address.
+If you need the functionality and there is already a `System` registered for that address, you can always deploy the bytecode as a new contract.
+
+### Upgrading systems
+
+The namespace owner can upgrade a `System`.
+This is a two-step process: deploy the contract for the new `System` and then call `registerSystem` with the same `ResourceId` as the old one and the new contract address.
+
+This upgrade process removes the old `System` contract's access to the namespace, and gives access to the new contract.
+Any access granted _manually_ to the `System` is not revoked, nor granted to the upgraded `System`.
+
+**Note:** You _should_ make sure to remove any such access.
+`System` access is based on the contract address, so somebody else could register a namespace they'd own, register the old `System` contract as a system in their namespace, and then abuse those permissions (if the `System` has code that can be used for that, of course).
+
+## Access control
+
+When you register a `System`, you can specify whether it is going to be private or public.
+
+- A public `System` has no access control checks, it can be called by anybody.
+  This is the main mechanism for user interaction with a MUD application.
+- A private `System` can only be called by entities that have access.
+  This access can be the result of:
+
+  - Access permission to the namespace in which the `System` is registered.
+  - Access permission specifically to the `System`.
+
+  Note that while code in private systems can only be called by authorized resources, by default those authorized resources include public systems in the same namespace operating on behalf of any user.
+  Vulnerabilities in private systems are _harder_ to exploit, but not necessarily impossible.
+
+## Root systems
+
+The `World` uses `call` for systems in other namespaces, but `delegatecall` for those in the root namespace (`bytes14(0)`).
+As a result, root systems have access to the `World` contract's storage.
+Because of this access, root systems use [the internal `StoreCore` methods](https://github.com/latticexyz/mud/blob/main/packages/store/src/StoreCore.sol), which are slightly cheaper than calling [the `IStore` methods](https://github.com/latticexyz/mud/blob/main/packages/store/src/IStore.sol) used by other systems.
+Note that the table libraries abstract this difference, so normally there is no reason to be concerned about it.
+
+Another effect of having access to the storage of the `World` is that root systems could, in theory, overwrite any information in any table regardless of access control.
+We recommend that the root namespace be used only when needed, and any `System` registered for it be properly audited.
+Of course, only the owner of the root namespace can register systems in the root namespace.
```

Comment:
Slight rephrasing. (Removed the part about auditing systems, sounded like we recommend _only_ root systems to be audited, but ideally all systems should be audited)
```suggestion
Only the [owner of the root namespace](./namespaces-access-control) can register root systems.
We recommend to only use the root namespace when strictly necessary.
```